scripts/deploy-backend.sh

```diff
@@ -0,0 +1,99 @@
+
+#!/bin/sh
+
+
+# The deployment strategy you wish to employ ( rolling update or setting up a new environment)
+DEPLOY_STRATEGY=${1}
+
+# The environment in which you want to execute these commands
+DEPLOY_ENV=${2}
+
+#The application name  defined via the manifest yml for the frontend
+CGHOSTNAME_BACKEND=${3}
+
+#The Github Branch triggered to execure this script if triggered in circleci
+CIRCLE_BRANCH=${4}
+
+echo DEPLOY_STRATEGY: $DEPLOY_STRATEGY
+echo DEPLOY_ENV=$DEPLOY_ENV
+echo BACKEND_HOST: $CGHOSTNAME_BACKEND
+echo CIRCLE_BRANCH=$CIRCLE_BRANCH
+
+#Helper method to generate JWT cert and keys for new environment
+generate_jwt_cert() 
+{
+	  echo "regenerating JWT cert/key"
+	  yes 'XX' | openssl req -x509 -newkey rsa:4096 -keyout key.pem -out cert.pem -days 365 -nodes -sha256
+	  cf set-env $CGHOSTNAME_BACKEND JWT_CERT "$(cat cert.pem)"
+	  cf set-env $CGHOSTNAME_BACKEND JWT_KEY "$(cat key.pem)"
+
+	  # make sure that we have something set that you can later override with the
+	  # proper value so that the app can start up
+	  if cf e $CGHOSTNAME_BACKEND | grep -q OIDC_RP_CLIENT_ID ; then
+		    echo OIDC_RP_CLIENT_ID already set up
+	  else
+		    echo "once you have gotten your client ID set up with login.gov, you will need to set the OIDC_RP_CLIENT_ID to the proper value"
+		    echo "you can do this by running: cf set-env tdp-backend OIDC_RP_CLIENT_ID 'your_client_id'"
+		    echo "login.gov will need this cert when you are creating the app:"
+		    cat cert.pem
+		    cf set-env $CGHOSTNAME_BACKEND OIDC_RP_CLIENT_ID "XXX"
+	  fi
+}
+
+update_backend()
+{
+    cd tdrs-backend
+	  if [ "$1" = "rolling" ] ; then
+		    # Do a zero downtime deploy.  This requires enough memory for
+		    # two apps to exist in the org/space at one time.
+		    cf push $CGHOSTNAME_BACKEND --no-route -f manifest.yml  --strategy rolling || exit 1
+	  else
+		    cf push $CGHOSTNAME_BACKEND --no-route -f manifest.yml
+		    # set up JWT key if needed
+		    if cf e $CGHOSTNAME_BACKEND | grep -q JWT_KEY ; then
+		        echo jwt cert already created
+		    else
+		        generate_jwt_cert
+	      fi
+	  fi
+	  cf map-route $CGHOSTNAME_BACKEND app.cloud.gov --hostname "$CGHOSTNAME_BACKEND"
+    cd ..
+}
+
+bind_backend_to_services() {
+		cf bind-service $CGHOSTNAME_BACKEND tdp-django-static-sandbox
+		cf bind-service $CGHOSTNAME_BACKEND tdp-storage-sandbox
+		cf bind-service $CGHOSTNAME_BACKEND tanf-storage
+		cf bind-service $CGHOSTNAME_BACKEND tdp-db
+
+    bash ./scripts/set-backend-env-vars.sh $CGHOSTNAME_BACKEND
+
+		cf restage $CGHOSTNAME_BACKEND
+}
+
+
```

Comment:
Previously we only had 2 deployment strategies, and we could not deploy new instances of the app with out manually setting connecting services and populating the environment.

Now this script can be used to bind services, set up an app instance from scratch, rebuild the app from scratch, and still do the original two deployment mechanisms of rolling/default.